RecipeApp/Modules/RecipeBook/ViewModel/RecipeBookViewModel.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  RecipeBookViewModel.swift
+//  RecipeApp
+//
+//  Created by Hadley Igoe on 2019-12-17.
+//  Copyright © 2019 Hadley Igoe. All rights reserved.
+//
+
+import Foundation
+
+class RecipeBookViewModel {
+    
+    var favoriteService = FavoriteService.sharedService
```

Comment:
Using a singleton is definitely a good way to prevent having copies of the service. I would still prefer to see it instantiated outside of the view model and passed in via its initializer. There's no way to test the view model this way (no way to set up the service to return specific data).